.ci/pipelines/e2e-tests-custom.Jenkinsfile

```diff
@@ -0,0 +1,103 @@
+// This library overrides the default checkout behavior to enable sleep+retries if there are errors
+// Added to help overcome some recurring github connection issues
+@Library('apm@current') _
+
+def failedTests = []
+def lib
+
+pipeline {
+
+    agent {
+        label 'linux'
+    }
+
+    options {
+        timeout(time: 600, unit: 'MINUTES')
+    }
+
+    parameters {
+        string(name: "OPERATOR_IMAGE", defaultValue: "", description: "ECK Docker image")
+        string(name: "ELASTICSEARCH_IMAGE", defaultValue: "", description: "Elasticsearch image under test ")
+        string(name: "ELASTICSEARCH_VERSION", defaultValue: "", description: "Elasticsearch version under test")
+        string(name: "KIBANA_IMAGE", defaultValue: "", description: "Kibana image under test ")
+        string(name: "KIBANA_VERSION", defaultValue: "", description: "Kibana version under test")
+        string(name: "ELASTIC_AGENT_IMAGE", defaultValue: "", description: "Elastic Agent image under test ")
+        string(name: "ELASTIC_AGENT_VERSION", defaultValue: "", description: "Elastic Agent version under test")
+        string(name: "ENTERPRISE_SEARCH_IMAGE", defaultValue: "", description: "Enterprise Search image under test ")
+        string(name: "ENTERPRISE_SEARCH_VERSION", defaultValue: "", description: "Enterprise Search version under test")
+        string(name: "ELASTIC_MAPS_SERVER_IMAGE", defaultValue: "", description: "Elastic Maps Server image under test ")
+        string(name: "ELASTIC_MAPS_SERVER_VERSION", defaultValue: "", description: "Elastic Maps Server version under test")
+        // Beats images are not supported here we would need 12 additional inputs to accomplish that
+        // APM Server is also not supported here given that is deprecated in favour of the Elastic Agent APM integration
+        string(name: "TESTS_MATCH", defaultValue: "^Test", description: "Regular expression to select which test cases to run")
```

Comment:
Should we allow specifying a tag instead/in addition to?

Reply:
You mean Go build tags to be able to run any test tanged with `agent` for example? Sure.

Reply:
> You mean Go build tags to be able to run any test tanged with `agent` for example? Sure.

Yes, thanks!